accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,475 @@
+* **System.Security.AccessControl.dll**
+* **System.Security.Cryptography.Cng.dll**
+* **System.Security.Cryptography.ProtectedData.dll**
+* **System.Security.Principal.Windows.dll**
+* **System.ServiceProcess.ServiceController.dll**
+* **System.Threading.AccessControl.dll**
+* **System.Threading.Overlapped.dll**
+
+## Specific APIs
+
+### System
+
+> **OPEN** Should we treat those as Windows-only or should we consider them as
+> partially portable?
+
+* **Console**
+    - `Beep(Int32, Int32)`
+    - `get_CapsLock()`
+    - `get_CursorVisible()`
+    - `get_NumberLock()`
+    - `get_Title()`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32)`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32, Char, ConsoleColor, ConsoleColor)`
+    - `set_BufferHeight(Int32)`
+    - `set_BufferWidth(Int32)`
+    - `set_CursorSize(Int32)`
+    - `set_WindowHeight(Int32)`
+    - `set_WindowLeft(Int32)`
+    - `set_WindowTop(Int32)`
+    - `set_WindowWidth(Int32)`
+    - `SetBufferSize(Int32, Int32)`
+    - `SetWindowPosition(Int32, Int32)`
+    - `SetWindowSize(Int32, Int32)`
+
+### System.Configuration
+
+> **OPEN** Should we mark the entire type?
+
+* **DpapiProtectedConfigurationProvider**
+    - `Decrypt(XmlNode)`
+    - `Encrypt(XmlNode)`
+
+### System.Diagnostics
+
+* **Process**
+    - `GetProcessById(Int32, String)`
+    - `GetProcesses(String)`
+    - `GetProcessesByName(String, String)`
+    - `set_MaxWorkingSet(IntPtr)`
+    - `set_MinWorkingSet(IntPtr)`
+    - `Start(String, String, SecureString, String)`
+    - `Start(String, String, String, SecureString, String)`
+* **ProcessStartInfo**
+    - `get_Domain()`
+    - `get_LoadUserProfile()`
+    - `get_Password()`
+    - `get_PasswordInClearText()`
+    - `set_Domain(String)`
+    - `set_LoadUserProfile(Boolean)`
+    - `set_Password(SecureString)`
+    - `set_PasswordInClearText(String)`
+* **ProcessThread**
+    - `set_PriorityLevel(ThreadPriorityLevel)`
+    - `set_ProcessorAffinity(IntPtr)`
+
+### System.Drawing
+
+> **OPEN** Should we mark the entire type?
+
+* **Graphics**
+    - `CopyFromScreen(Int32, Int32, Int32, Int32, Size)`
+    - `CopyFromScreen(Int32, Int32, Int32, Int32, Size, CopyPixelOperation)`
+    - `CopyFromScreen(Point, Point, Size)`
+    - `CopyFromScreen(Point, Point, Size, CopyPixelOperation)`
+
+### System.IO.MemoryMappedFiles
+
+* **MemoryMappedFile**
+    - `CreateFromFile(FileStream, String, Int64, MemoryMappedFileAccess, HandleInheritability, Boolean)`
+    - `CreateFromFile(String)`
+    - `CreateFromFile(String, FileMode)`
+    - `CreateFromFile(String, FileMode, String)`
+    - `CreateFromFile(String, FileMode, String, Int64)`
+    - `CreateFromFile(String, FileMode, String, Int64, MemoryMappedFileAccess)`
+    - `CreateNew(String, Int64)`
+    - `CreateNew(String, Int64, MemoryMappedFileAccess)`
+    - `CreateNew(String, Int64, MemoryMappedFileAccess, MemoryMappedFileOptions, HandleInheritability)`
```

Comment:
All of the above Create* methods are fully supported on unix as long as the supplied string is null.  This is an example of where the currently proposed mechanism is insufficient.  These should not be marked as Windows-only.

Reply:
I noted down that we would remove the `Create` methods but leave the `Open` methods in the list here. I am also removing the `CreateOrOpen` methods since they also look to unconditionally throw on Unix. Please let me know if I've misunderstood those though.

Reply:
> I am also removing the CreateOrOpen methods since they also look to unconditionally throw on Unix.

CreateFromFile and CreateNew are not Windows-only. CreateOrOpen and OpenExisting are Windows-only.

Reply:
OK got it; adding `CreateOrOpen` back into the list. Thanks.